### PR TITLE
[BetterPhpDocParser] Handle illegal offset type of ConstExprIntegerNode on ArrayParser

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes https://github.com/rectorphp/rector/issues/6751

### CODE REVIEW

**Comment by samsonasik** [comment]
**Before**

Screen Shot 2021-10-19 at 16 28 46


**After**

Screen Shot 2021-10-19 at 16 28 58

**Comment by Bilge** [comment]
I can't really try it due to the weird way these packages are set up ("rector/rector": "dev-skip-illegal-offset-type" doesn't actually exist). I also can't switch it out for "rector/rector-src" because that requires PHP 8 and I'm on 7.4.

**Comment by samsonasik** [comment]
@Bilge  can  you manually copy the patch and try ?

**Comment by Bilge** [comment]
@samsonasik Manually editing the source seems to work 👍🏻

**Comment by samsonasik** [comment]
@Bilge thank you for verify, I am merging it ;)

**Comment by Bilge** [comment]
When will it be tagged?

File: rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/skip_illegal_offset_type.php.inc [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+     * @Assert\All(
+     *     constraints={
+     *         121 = "test"
+     *     }
+     * )
+     */
+    public $all;
+}
+
+?>
```
Comment (Bilge) [inline rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/skip_illegal_offset_type.php.inc:19]: Isn't it widely regarded as bad practice to end PHP-only files this way? That is, this line should be removed.

Comment (samsonasik) [reply rules-tests/Php80/Rector/Class_/AnnotationToAttributeRector/Fixture/skip_illegal_offset_type.php.inc:19]: Yes, a copy paste mistake :), thanks for the check.

**Comment by samsonasik** [comment]
You can use "rector/rector": "dev-main" while waiting for next release

**Comment by samsonasik** [comment]
0.11.59 released https://github.com/rectorphp/rector/releases/tag/0.11.59

**Pull request merged**